app/scaffold-controllers/drawerController.js

```diff
@@ -78,51 +68,42 @@ function(
 
         var initNavigationItemsPromise = Promise.join(
             initLeftMenuPromise,
-            constructTabItemsPromise,
             counterBadgeControllerPromise,
             cartEventHandlerPromise,
             errorControllerPromise,
             initNavigationItems);
 
-        return Promise.all([initNavigationItemsPromise]).then(function() {
-            return Promise.join(initNavigationItemsPromise, webViewPromise, function(drawer, leftMenu) {
-                return new DrawerController(drawer, leftMenu);
-            });
+        return Promise.join(initNavigationItemsPromise, webViewPromise, function(drawer, leftMenu) {
+            leftMenu.disableDefaultNavigationHandler();
+            drawer.setContentView(drawer.navigationController.viewPlugin);
+            return new DrawerController(drawer, leftMenu);
         });
     };
 
-    DrawerController.prototype.selectItem = function(itemId) {
-        if (this.activeItemId !== itemId) {
-            if (this.activeItemId) {
-                this.drawer.itemControllers[this.activeItemId].isActive = false;
-            }
-            this.drawer.setContentView(this.drawer.itemViews[itemId]);
-            var selectedItem = this.drawer.itemControllers[itemId];
-            selectedItem.isActive = true;
-            this.activeItemId = itemId;
+    DrawerController.prototype.navigateToNewRootView = function(url, title) {
+        // Strip away 'file://' from relative urls coming from anchor tags
+        if (url.indexOf('file://') > -1) {
+            url = url.replace('file://', BaseConfig.baseURL);
         }
-    };
-
-    DrawerController.prototype._bindEvents = function() {
-        var self = this;
-
-        this.leftMenu.on('menuItemSelected', function(data) {
-            self.drawer.hideLeftMenu();
-            self.selectItem(data.id);
-        });
-    };
-
-    DrawerController.prototype.navigateActiveItem = function(url) {
-        this.drawer.itemControllers[this.activeItemId].navigate(url);
+        this.drawer.navigationController.navigateMainViewToNewRoot(url, title);
+        this.drawer.hideLeftMenu();
     };
 
     DrawerController.prototype.backActiveItem = function() {
-        this.drawer.itemControllers[this.activeItemId].back();
+        var activeItem = this.drawer.navigationController;
```

Comment:
Is it for readability that this is done across two lines? Could it be combined into one?